[12.0][MIG][ADD] openupgrade migrations

vvrossem opened the pull request:
# THIS PR IS NOT TO BE MERGED
The `12.0-openupgrade` branch is used for OpenUpgrade migrations.

### easy_my_coop
This solves the critical error `duplicate key value violates unique constraint "res_groups_name_uniq"` raised when using openupgrade. 
This error is due to the fact that `group_energiris_user` has been renamed as `group_easy_my_coop_user` and `group_energiris_manager` in `group_easy_my_coop_manager`

### website_recaptcha_reloaded
9.0 -> 12.0

vvrossem committed:
[ADD] easy_my_coop: openupgrade.rename_xmlids()

vvrossem committed:
[ADD] website_recaptcha_reloaded: pre-migration.py

vvrossem committed:
[WIP] easy_my_coop: migration script

houssine78 commented:
@vvrossem I think that it has been used for two or three migration. Can we go for merging it ?

vvrossem commented:
@remytms ok to merge ?

houssine78 commented:
@remytms can we merge this ?

remytms commented:
We should no merge this PR as it will force to install `openupgradelib` on production servers.

So do no merge this PR. It stands here to not remove the 12.0-openugrade branch.

carmenbianca committed:
[FIX] easy_my_coop: Set mail.template.easy_my_coop field to True on migration

Previously, for some reason, these did not default to True, or somehow
lost their truthiness during migration from v9 to v12. Whatever the
case, this should make sure that doesn't happen.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

huguesdk committed:
[FIX] fix mail templates migration script

huguesdk committed:
Merge branch '12.0-mail-template-migration' into 12.0-openupgrade

huguesdk committed:
Merge branch '12.0' into 12.0-openupgrade

huguesdk committed:
[IMP] update and add migrations scripts

huguesdk committed:
Merge branch '12.0' into 12.0-openupgrade
